src/backend/distributed/transaction/transaction_management.c

```diff
@@ -262,6 +266,11 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 				AfterXactConnectionHandling(true);
 			}
 
+			if (MetadataSyncOnCommit)
+			{
+				TriggerMetadataSync(MyDatabaseId);
```

Comment:
would be good to add a comment on why this is here (and needs to be in XACT_EVENT_COMMIT, and not earlier).